### PR TITLE
Task10

### PR DESCRIPTION
**Pull request by paulius27**
Add a service that will save weather history.

### CODE REVIEW

File: WeatherApp/BL/WeatherHistoryService.cs [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        private readonly IWeatherService _weatherService;
+
+        public WeatherHistoryService(ILogger<WeatherHistoryService> logger, IWeatherHistoryRepository weatherHistoryRepository, IWeatherService weatherService) 
+        {
+            _logger = logger;
+            _weatherHistoryRepository = weatherHistoryRepository;
+            _weatherService = weatherService;
+        }
+
+        public async Task UpdateWeatherHistory(IEnumerable<string> cityNames)
```
Comment (MaximGomon) [inline WeatherApp/BL/WeatherHistoryService.cs]: try to use params string[]

File: WeatherApp/DAL/Models/WeatherHistoryEntry.cs [hunk 2]
```diff
@@ -0,0 +1,15 @@
+﻿using System;
+
+namespace DAL.Models
+{
+    public class WeatherHistoryEntry
+    {
+        public int Id { get; set; }
+
+        public string CityName { get; set; }
```
Comment (MaximGomon) [inline WeatherApp/DAL/Models/WeatherHistoryEntry.cs]: use MaxLenght attribute

File: WeatherApp/DAL/Migrations/20230216102051_InitialCreate.cs [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        /// <inheritdoc />
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+            migrationBuilder.CreateTable(
+                name: "WeatherHistory",
+                columns: table => new
+                {
+                    Id = table.Column<int>(type: "int", nullable: false)
+                        .Annotation("SqlServer:Identity", "1, 1"),
+                    CityName = table.Column<string>(type: "nvarchar(max)", nullable: false),
```
Comment (MaximGomon) [inline WeatherApp/DAL/Migrations/20230216102051_InitialCreate.cs:20]: Think about Cities table and FK

File: WeatherApp/WebAPI/Jobs/UpdateWeatherHistoryJob.cs [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+        private readonly IWeatherHistoryService _weatherHistoryService;
+
+        public UpdateWeatherHistoryJob(IWeatherHistoryService weatherHistoryRepository)
+        {
+            _weatherHistoryService = weatherHistoryRepository;
+        }
+
+        public async Task Execute(IJobExecutionContext context)
+        {
+            var cityNames = (IEnumerable<string>)context.MergedJobDataMap["cityNames"];
```
Comment (MaximGomon) [inline WeatherApp/WebAPI/Jobs/UpdateWeatherHistoryJob.cs]: Check for a null. move literal of "cityNames" to the some constants

File: WeatherApp/BL/WeatherHistoryService.cs (hunk 1 above)
Comment (paulius27) [reply WeatherApp/BL/WeatherHistoryService.cs]: Params string[] is now used for cityNames.

File: WeatherApp/DAL/Models/WeatherHistoryEntry.cs (hunk 2 above)
Comment (paulius27) [reply WeatherApp/DAL/Models/WeatherHistoryEntry.cs]: Added MaxLength attribute.

File: WeatherApp/WebAPI/Jobs/UpdateWeatherHistoryJob.cs (hunk 4 above)
Comment (paulius27) [reply WeatherApp/WebAPI/Jobs/UpdateWeatherHistoryJob.cs]: Added null check and moved "cityNames" to separate constants file.

File: WeatherApp/DAL/Migrations/20230216102051_InitialCreate.cs (hunk 3 above)
Comment (paulius27) [reply WeatherApp/DAL/Migrations/20230216102051_InitialCreate.cs:20]: Added Cities table and now use city Id as foreign key.

**Review by MaximGomon — APPROVED** [review]

**Pull request merged**